src/utils/CalendarUtils/CalendarUtils.js

```diff
@@ -37,30 +38,67 @@ export const elementExists = selector => {
   return javaScriptUtils.isDefined(element);
 };
 
-export const filterContributionDays = (dailyData, timeframe) => {
-  const isDay = dailyData.attributes.class === "day";
-
-  // Weekdays and months displayed around the calendar should be disregarded.
-  if (!isDay) {
-    return false;
-  }
-
+export const filterByTimeframe = (dailyDataWithContributions, timeframe) => {
   if (!timeframe.start && !timeframe.end) {
-    return true;
+    return dailyDataWithContributions;
   }
 
-  const contributionsDate = new Date(dailyData.attributes["data-date"]);
+  const filtered = Object.entries(dailyDataWithContributions).filter(
+    ([date]) => {
+      const convertedDate = new Date(date);
 
-  if (timeframe.start && !timeframe.end) {
-    return contributionsDate >= new Date(timeframe.start);
-  }
+      if (timeframe.start && !timeframe.end) {
+        return convertedDate >= new Date(timeframe.start);
+      }
 
-  if (!timeframe.start && timeframe.end) {
-    return contributionsDate <= new Date(timeframe.end);
-  }
+      if (!timeframe.start && timeframe.end) {
+        return convertedDate <= new Date(timeframe.end);
+      }
 
-  return (
-    contributionsDate >= new Date(timeframe.start) &&
-    contributionsDate <= new Date(timeframe.end)
+      return (
+        convertedDate >= new Date(timeframe.start) &&
+        convertedDate <= new Date(timeframe.end)
+      );
+    }
+  );
+
+  return filtered.reduce(
```

Comment:
We may rather [`Object.fromEntries()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries) to transform the array of arrays into an object. My concerns are that [certain versions of browsers](https://caniuse.com/#search=fromEntries) don't yet support it(e.g. IE, what a surprise) and the tests failed because the runner couldn't recognize this function for some reason.

Reply:
Speaking of browser support: https://medium.com/hackernoon/introducing-immer-immutability-the-easy-way-9d73d8f71cb3 (see "No proxies?" chapter).

Reply:
Looking at the "usage relative" chart it's ok t use imo. However, wouldn't our build process take care of things like these? Don't we transpile the code to work from most browsers?

> see "No proxies?" chapter
I don't know what proxies are in this context.

Reply:
> However, wouldn't our build process take care of things like these?

`Object.fromEntries()` is *not* transpiled after running either the `lib` or `distify` script.

> Don't we transpile the code to work from most browsers?

You're right, the target browsers are explicitly set:

https://github.com/c-hive/team-contribution-calendar/blob/f6ce3a54aab963a1b4584cdc574e39fee6fa599a/package.json#L86-L94

It should answer why it's not transpiled.

As for the error raised by tests: switching to a recent version of node(>=12.4.0) could resolve the issue (https://node.green/#ES2019).

Reply:
> switching to a recent version of node(>=12.4.0)

Would it make sense to specify the minimum node version?

Reply:
☝️ https://github.com/c-hive/team-contribution-calendar/pull/70/commits/4096ca7b250ee1f950a62b0a18633b48a66ee07e

Reply:
:+1:

Reply:
Would it make sense to set the `browserslist` to a broader one and include IE? If there's no noticeable downside we could do that.

Reply:
> Would it make sense to set the browserslist to a broader one and include IE?

Yea, it'd. But `Object.entries()` is still *not* transpiled for some reason.

https://github.com/babel/babel/issues/8362#issuecomment-407086767

It says we should use a "polyfill" to transpile methods.

https://babeljs.io/docs/en/babel-polyfill

> 🚨 As of Babel 7.4.0, this package has been deprecated in favor of directly including core-js/stable (to polyfill ECMAScript features) and regenerator-runtime/runtime (needed to use transpiled generator functions):

What this actually indicates is that we can directly import the "only-required-for-you polyfills" at the top of the entry point:

```js
import "core-js/stable";
import "regenerator-runtime/runtime";
```

One another, and way more preferred workaround to set up the polyfill, is:

https://github.com/zloirock/core-js#babelpreset-env

> @babel/preset-env has useBuiltIns option, which optimizes working with global version of core-js. With useBuiltIns option, you should also set corejs option to used version of core-js, like corejs: '3.6'.

So something like this:

```
// .babelrc

{
   "presets": [
      "@babel/preset-env,
      {
         // adds to the top of each file import of polyfills for features used in this file and not supported by target environments, so for
         "useBuiltIns": "entry"
      }
   ]
}
```

One another thing I've "observed" is that the `browserslist` entry isn't picked up by Babel in production whatsoever:

https://github.com/c-hive/team-contribution-calendar/blob/f6ce3a54aab963a1b4584cdc574e39fee6fa599a/package.json#L87

https://github.com/browserslist/browserslist#best-practices

> There is a defaults query, which gives a reasonable configuration for most users:

I'd fall back to their reasonable set of defaults for both development and production.

Reply:
> One another thing I've "observed" is that the browserslist entry isn't picked up by Babel in production whatsoever:

Ahh, now I see. So IE 11 is excluded from the list of browsers because it doesn't even support e.g. [async functions](https://caniuse.com/#search=async).

https://github.com/babel/babel/issues/9849#issuecomment-487040428

> To fix this, simply use regenerator-runtime, following the instructions in its README to ensure regeneratorRuntime is made available globally.

I don't think it's worth adding yet another dependency to make the package available for a relatively small subset of users (1.43%).

Reply:
https://github.com/c-hive/team-contribution-calendar/pull/70/commits/a120cad1391ec4f2c750fcd9b01e19a6f9ffc095

Reply:
> One another thing I've "observed" is that the browserslist entry isn't picked up by Babel in production whatsoever:

One more thing to add to this is that `browserslist` will fall back to the `defaults` if the target environment isn't specified:

> If none of them is declared, Browserslist will firstly look for production queries and then use defaults.

Reply:
> I don't think it's worth adding yet another dependency to make the package available for a relatively small subset of users (1.43%).

These are all dev dependencies though, right? If there's a configuration that isn't too complicated that allows us to have a bigger reach, let's use it. If not, 1.4% is not a bug issue.

Reply:
> not dead with a usage number like > 0.2%

This sounds reasonable to me.

Reply:
I don't get the fallbacks in different envronments. "target environment isn't specified" => How do you specify it? Why isn't it specified?

Reply:
> These are all dev dependencies though, right? If there's a configuration that isn't too complicated that allows us to have a bigger reach, let's use it. If not, 1.4% is not a bug issue.

Yep, AFAICU it's merely a dev dependency and a one-line import at the top of the entry file.

> How do you specify it?

Target environment == "target browsers". These can be defined one-by-one by editing the `browserslist` entry of `package.json`. In our case, the browsers are *only* specified for development (i.e. `yarn run dev`). In the case of production, however, since there's no `production` entry, it falls back to the [reasonable set of defaults](https://browserl.ist/).

As for "Why isn't it specified?": https://github.com/c-hive/team-contribution-calendar/pull/44#discussion_r318946538.

Reply:
To cut a long story short, I'd say the setup is fine as-is, let's just support IE11.

Reply:
> let's just support IE11.

Ignore this. The setup is already fine as-is.

https://github.com/c-hive/team-contribution-calendar/blob/f6ce3a54aab963a1b4584cdc574e39fee6fa599a/.babelrc#L9-L16

We're already using `@babel/plugin-transform-runtime` to transpile generators.

Reply:
> Target environment == "target browsers". These can be defined one-by-one by editing the browserslist entry of package.json. In our case, the browsers are only specified for development (i.e. yarn run dev). In the case of production, however, since there's no production entry, it falls back to the reasonable set of defaults.

Got it, thanks. Didn't know there's a difference between prod and dev. We should, first of all, have the same config for both where possible. Secondly, we should support the widest audience reasonably possible.

>  The setup is already fine as-is.

I'd propose to use "not dead with a usage number like > 0.2%" in both dev and prod. Is that easy to implement?

Reply:
> I'd propose to use "not dead with a usage number like > 0.2%" in both dev and prod. Is that easy to implement?

https://browserl.ist/?q=last+2+versions+and+not+dead+and+%3E+0.2%25

AFAICT it's as simple as using `@babel/plugin-transform-runtime` for development as well.

Reply:
https://github.com/c-hive/team-contribution-calendar/pull/70/commits/817acbfecf40de7af5a28009f525e179a2923349